sycl/doc/extensions/proposed/sycl_ext_intel_fpga_kernel_interface_properties.asciidoc

```diff
@@ -0,0 +1,296 @@
+  using value_t = property_value<register_map_interface_key,
+    std::integral_constant<register_map_interface_options_enum, option>>;
+};
+
+struct pipelined_key {
+  template <bool pipeline_directive, typename initiation_interval>
+  using value_t = property_value<pipelined_key, pipeline_directive,
+   initiation_interval>;
+};
+
+template <streaming_interface_options_enum option>
+inline constexpr streaming_interface_key::value_t<option> streaming_interface;
+
+inline constexpr streaming_interface_key::value_t<
+  streaming_interface_options_enum::accept_downstream_stall>
+    streaming_interface_accept_downstream_stall;
+
+inline constexpr streaming_interface_key::value_t<
+  streaming_interface_options_enum::remove_downstream_stall>
+    streaming_interface_remove_downstream_stall;
+
+inline constexpr register_map_interface_key::value_t<
+  register_map_interface_options_enum::wait_for_done_read>
+    register_map_interface_wait_for_done_read;
+
+inline constexpr register_map_interface_key::value_t<
+  register_map_interface_options_enum::do_not_wait_for_done_read>
+    register_map_interface_do_not_wait_for_done_read;
+
+inline constexpr register_map_interface_key::value_t<option>
+  register_map_interface;
+
+template<int initiation_interval_value>
+inline constexpr initiation_interval_key::value_t<initiation_interval_value>
+  initiation_interval;
+
+template<bool pipeline_directive, typename initiation_interval>
+inline constexpr pipelined_key::value_t<pipeline_directive,
+  initiation_interval> pipelined;
+
+} // namespace experimental
+} // namespace oneapi
+} // namespace ext
+} // namespace sycl
+```
+
+|===
+|Property|Description
+|`streaming_interface`
+|The `streaming_interface` property adds the requirement that the kernel must
+ have dedicated ports for input / output signals. This applies for both
+ control, and kernel argument data signals. The following values are supported:
+
+ * `accept_downstream_stall`: Provides the compiler the guarantee that
+    downstream logic will back-pressure the kernel.
+
+ * `remove_downstream_stall`: Provides the compiler the guarantee that
+    downstream logic will not back-pressure the kernel.
+
+ If the `streaming_interface` property is not specified, the default behavior is
+ equivalent to one of the values listed above, but the choice is implementation
+ defined.
+
+ The following properties have been provided for convenience:
+ `streaming_interface_accept_downstream_stall`,
+ `streaming_interface_remove_downstream_stall`.
+
+|`register_map_interface`
+|The `register_map_interface` property adds the requirement that the kernel must have
+ its input / output control and kernel argument data signals placed in a shared
+ Control and Status Register (CSR) map. The following values are supported:
+
+ * `wait_for_done_read`: Provides the compiler the guarantee that downstream
+    logic will back-pressure the kernel.
+
+ * `do_not_wait_for_done_read`: Provides the compiler the guarantee that
```

Comment:
This one is more of a guarantee but it still seems like the wrong choice of words.  The user is telling the compiler that it can build hardware that doesn't backpressure the kernel.  I.e. that the user's system is either tolerant to dropped dones or that they are going to throttle the kernel themselves sufficiently to ensure that no done are dropped.